### ISSUE TITLE
Autoscaling not working for the same label

### ISSUE THREAD

**Issue by andreroggeri** [jenkinsci/ec2-fleet-plugin#261]
## Issue Details

**Describe the bug**
The fleet is not scaling when more jobs arrive for the label `android`. The fleet agent has two labels: `android` and `appium`.
(For `appium` label it scales immediately after the job stays on the queue)

Looking that the logs, its seems that the demand count is wrong (negative demand) and the machine count seems weird too (Logs shows that we have 4 computers, but there is only 2 instances on the fleet)

### 1 Job on the queue with label `android`
```
AndroidAgent [android appium] start
Mar 03, 2021 9:44:50 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] fleet instances: [i-0056687a27c1b205a]
Mar 03, 2021 9:44:50 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] described instances: [i-0056687a27c1b205a]
Mar 03, 2021 9:44:50 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] jenkins nodes: [i-0056687a27c1b205a]
Mar 03, 2021 9:44:58 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [android]: currentDemand -1 availableCapacity 2 (availableExecutors 0 connectingExecutors 0 plannedCapacitySnapshot 2 additionalPlannedCapacity 0)
Mar 03, 2021 9:44:58 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [android]: currentDemand is less than 1, not provisioning
Mar 03, 2021 9:45:00 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] start
Mar 03, 2021 9:45:00 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] fleet instances: [i-0056687a27c1b205a]
Mar 03, 2021 9:45:00 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] described instances: [i-0056687a27c1b205a]
Mar 03, 2021 9:45:00 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] jenkins nodes: [i-0056687a27c1b205a]
```

### 2 Jobs on the queue (One `android` and another `appium`)
```
Mar 03, 2021 9:46:20 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] jenkins nodes: [i-0056687a27c1b205a]
Mar 03, 2021 9:46:28 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [android]: currentDemand -1 availableCapacity 2 (availableExecutors 0 connectingExecutors 0 plannedCapacitySnapshot 2 additionalPlannedCapacity 0)
Mar 03, 2021 9:46:28 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [android]: currentDemand is less than 1, not provisioning
Mar 03, 2021 9:46:28 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [appium]: currentDemand 1 availableCapacity 0 (availableExecutors 0 connectingExecutors 0 plannedCapacitySnapshot 0 additionalPlannedCapacity 0)
Mar 03, 2021 9:46:28 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [appium]: cloud AndroidAgent can provision for this label
Mar 03, 2021 9:46:28 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] excessWorkload 1
Mar 03, 2021 9:46:28 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] to provision = 1
Mar 03, 2021 9:46:30 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] start
Mar 03, 2021 9:46:30 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] Update fleet target capacity to 2
Mar 03, 2021 9:46:30 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] fleet instances: [i-0056687a27c1b205a]
Mar 03, 2021 9:46:30 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] described instances: [i-0056687a27c1b205a]
```

### After scaling
```
Mar 03, 2021 9:49:08 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [appium]: currentDemand 0 availableCapacity 1 (availableExecutors 0 connectingExecutors 0 plannedCapacitySnapshot 1 additionalPlannedCapacity 0)
Mar 03, 2021 9:49:08 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [appium]: currentDemand is less than 1, not provisioning
Mar 03, 2021 9:49:10 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] start
Mar 03, 2021 9:49:10 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] fleet instances: [i-086da525597fae4c6, i-0056687a27c1b205a]
Mar 03, 2021 9:49:10 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] described instances: [i-086da525597fae4c6, i-0056687a27c1b205a]
Mar 03, 2021 9:49:10 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
AndroidAgent [android appium] jenkins nodes: [i-086da525597fae4c6, i-0056687a27c1b205a]
Mar 03, 2021 9:49:10 AM INFO com.amazon.jenkins.ec2fleet.EC2FleetOnlineChecker run
i-086da525597fae4c6 connected, resolve planned node
Mar 03, 2021 9:49:18 AM INFO hudson.slaves.NodeProvisioner lambda$update$6
FleetNode-0 provisioning successfully completed. We have now 4 computer(s)
Mar 03, 2021 9:49:18 AM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
```
## Environment Details

**Plugin Version?**
2.3.2

**Jenkins Version?**
2.275

**Spot Fleet or ASG?**
Spot Fleet

**Label based fleet?**
Yes

**Using No Delay Provision Strategy?**
Yes

**Linux or Windows?**
Yes

**Anything else unique about your setup?**
No

**Comment by haugenj** [comment]
:wave: @andreroggeri 

The plugin isn't scaling because `plannedCapacitySnapshot` is positive, so instances were already _supposed_ to be requested. We've had problems with this before though, such as in #172

a couple questions:
- Have you seen this happen frequently? The most common scenario we saw this before was when changes to the plugin were made while there were pending jobs in the queue
- If you restart Jenkins, does this still occur?
- Do you have a full set of logs you could send me? I tried to fix this problem before in #236 by adding a timeout, I'm wondering if I need to revisit that

**Comment by andreroggeri** [comment]
Hi ! Thanks for the quick feedback.

> Have you seen this happen frequently? The most common scenario we saw this before was when changes to the plugin were made while there were pending jobs in the queue

Yes, its been like this for a while but its possible that I changed the plugin during fleet updates
>If you restart Jenkins, does this still occur?
I just restarted our server and it seems to be scaling correctly.

> Do you have a full set of logs you could send me? I tried to fix this problem before in #236 by adding a timeout, I'm wondering if I need to revisit that
I'm particularly interested in what the first NoDelayProvisionStrategy apply log is for the android label, and whether you ever see this line "scaling timeout reached, removing node from Jenkins's plannedCapacitySnapshot"
I'll keep an eye and if it starts to misbehave again I'll update here (Have to learn how to configure jenkins to keep logs :sweat_smile: ), but I don't remember seeing any scaling timeout message.
